render.c

```diff
@@ -36,9 +37,32 @@ static void move_to(cairo_t *cairo, double x, double y, int scale) {
 	cairo_move_to(cairo, x * scale, y * scale);
 }
 
-static void set_rectangle(cairo_t *cairo, double x, double y, double width, double height,
-		int scale) {
-	cairo_rectangle(cairo, x * scale, y * scale, width * scale, height * scale);
+static void rounded_rectangle(cairo_t *cairo, double x, double y, double width,
+                    double height, double radius) {
+	double degrees = M_PI / 180.0;
+
+	if (radius > width / 2) {
+		radius = 0;
+	}
+
+	if (radius > height / 2) {
+		radius = 0;
+	}
+
+	cairo_new_sub_path(cairo);
+	cairo_arc(cairo, x + width - radius, y + radius, radius, -90 * degrees, 0 * degrees);
+	cairo_arc(cairo, x + width - radius, y + height - radius, radius, 0 * degrees, 90 * degrees);
+	cairo_arc(cairo, x + radius, y + height - radius, radius, 90 * degrees, 180 * degrees);
+	cairo_arc(cairo, x + radius, y + radius, radius, 180 * degrees, 270 * degrees);
+	cairo_close_path(cairo);
+} 
+
+static void set_rounded_rectangle(cairo_t *cairo, double x, double y, double width, double height,
+		int scale, double radius) {
+	if (width != 0 && height != 0) {
+		rounded_rectangle(cairo, x * scale, y * scale, width * scale,
```

Comment:
This could just be

```c
if (width == 0 || height == 0) {
	return;
}
```

at the beginning of the function, no need to declare two functions